### DISCUSSION TITLE
`Ant build` command works but intelliJ does not recognize the whole project

### DISCUSSION THREAD

**Discussion by SaeidKazemi78** [Keyfactor/ejbca-ce#737] [Q&A] [answered]
I want to build ejbca-ce project and deploy it on wildfly , `ant build ` works properly and generates ejbca.ear file ,
but when i want to develop some changes , the IDE intelliJ does not recognize some modules , for instance :CaRestResourceSwagger at ejbca-rest-api module and it disrupts development process

please tell me what is your IDE to build , or guide me about how to resolve this on intelliJ ?  

@primetomas 

![Screenshot from 2024-12-11 11-04-02](https://github.com/user-attachments/assets/be0f9630-37ce-42c0-8c98-95abd7bffe92)

**Comment by primetomas** [comment]
Personally I use Eclipse, but others use other IDEs.

**Comment by dobicinaitis** [comment, marked as answer]
Hi!

You're encountering these issues because the project is imported as a Gradle project. Gradle support is still experimental and lacks most build targets/tasks, so you should use Ant instead. Unfortunately, the Gradle configuration files in the repository confuse IntelliJ, causing it to use them by default.

Here's what you need to do:

* Enable or install the Ant plugin in IntelliJ under ⚙️ ❯ `Plugins ...`.
* Close the project from `File` ❯ `Close Project`, then remove it from recent projects using the three-dot menu.
* Delete IntelliJ configuration files from the `ejbca-ce` directory (`rm -rf .idea *.iml`).
* Open the `ejbca-ce` directory in IntelliJ. When prompted to choose between Eclipse and Gradle configurations, select the "Eclipse project" option.
* Click the `Add Ant build file` button in the bottom-right pop-up notification.
* Navigate to `File` ❯ `Project Structure`, set the `Language Level` to `17`, and click `Apply`.
* Go to `Modules` ❯ ➕ ❯ `Import Module` and select the `ejbca-ce` project again.
* Choose `Import module from external model` ❯ `Eclipse` ❯ `Next` ❯ `Next` ❯ `Next` ❯ `Create`.
* Change the `Module SDK` to `17`.
* Click `OK`.
* Maybe restart IntelliJ.

After completing these steps, all imports should resolve. Happy coding! 🧑‍💻

P.S. Once Gradle is fully supported (no ETA yet), you should be able to import the project hassle-free just as you did initially.

**Reply by SaeidKazemi78** [reply]
Thanks a millioooon  for your comprehensive answer . i did my task on eclipse , there is no issue on eclipse IDE . but working with eclipse is much annoying for us.
